File: spring-reactive-02/learn-java/src/main/java/boki/learnjava/v2_filtersequence/c_skipext/SkipExtendedEx4.java
package boki.learnjava.v2_filtersequence.c_skipext;

import boki.learnjava.common.SampleData;
import boki.learnjava.utils.Logger;
import reactor.core.publisher.Flux;

/**
 * skipWhile 기본 예제
 *  - 파라미터로 입력되는 Predicate가 true가 될때까지 건너뜀
 */
public class SkipExtendedEx4 {
    public static void main(String[] args) {
        Flux
            .fromIterable(SampleData.btcTopPricesPerYear)
            .skipWhile(tuple -> tuple.getT2() < 10_000_000)
            .subscribe(tuple -> Logger.onNext(tuple.getT1(), tuple.getT2()));
    }
}

File: spring-reactive-02/learn-kt/src/main/kotlin/boki/learnkt/v2_filtersequence/c_skipext/SkipExtendedEx4.kt
package boki.learnkt.v2_filtersequence.c_skipext

import boki.learnjava.common.SampleData
import boki.learnkt.util.Logger
import reactor.core.publisher.Flux

/**
 * skipWhile 기본 예제
 *  - 파라미터로 입력되는 Predicate가 true가 될때까지 건너뜀
 */
fun main() {
    Flux
        .fromIterable(SampleData.btcTopPricesPerYear)
        .skipWhile { it.t2 < 10_000_000 }
        .subscribe { Logger.onNext(it.t1, it.t2) }
}